glide/src/androidTest/java/dev/chrisbanes/accompanist/glide/GlideTest.kt

```diff
@@ -0,0 +1,462 @@
+                            latch.countDown()
+                            // False so that Glide still invokes the Target
+                            return false
+                        }
+                    })
+                },
+                modifier = Modifier.preferredSize(128.dp, 128.dp),
+                onRequestCompleted = { results += it }
+            )
+        }
+
+        // Wait for the Glide request listener to release the latch
+        latch.await(5, TimeUnit.SECONDS)
+
+        composeTestRule.runOnIdle {
+            // And assert that we got a single successful result
+            assertThat(results).hasSize(1)
+            assertThat(results[0]).isInstanceOf(ImageLoadState.Success::class.java)
+        }
+    }
+
+    @Test
+    fun basicLoad_http() {
+        val latch = CountDownLatch(1)
+
+        composeTestRule.setContent {
+            GlideImage(
+                data = server.url("/image").toString(),
+                modifier = Modifier.preferredSize(128.dp, 128.dp).testTag(GlideTestTags.Image),
+                onRequestCompleted = { latch.countDown() }
+            )
+        }
+
+        // Wait for the onRequestCompleted to release the latch
+        latch.await(5, TimeUnit.SECONDS)
+
+        composeTestRule.onNodeWithTag(GlideTestTags.Image)
+            .assertIsDisplayed()
+            .assertWidthIsEqualTo(128.dp)
+            .assertHeightIsEqualTo(128.dp)
+    }
+
+    @Test
+    @SdkSuppress(minSdkVersion = 26) // captureToBitmap is SDK 26+
+    fun basicLoad_drawable() {
+        val latch = CountDownLatch(1)
+
+        composeTestRule.setContent {
+            GlideImage(
+                data = resourceUri(R.drawable.red_rectangle),
+                modifier = Modifier.preferredSize(128.dp, 128.dp).testTag(GlideTestTags.Image),
+                onRequestCompleted = { latch.countDown() }
+            )
+        }
+
+        // Wait for the onRequestCompleted to release the latch
+        latch.await(5, TimeUnit.SECONDS)
+
+        composeTestRule.onNodeWithTag(GlideTestTags.Image)
+            .assertWidthIsEqualTo(128.dp)
+            .assertHeightIsEqualTo(128.dp)
+            .assertIsDisplayed()
+            .captureToBitmap()
+            .assertPixels { Color.Red }
+    }
+
+    @OptIn(ExperimentalCoroutinesApi::class)
+    @Test
+    @SdkSuppress(minSdkVersion = 26) // captureToBitmap is SDK 26+
+    fun basicLoad_switchData() {
+        val loadCompleteSignal = Channel<Unit>(Channel.UNLIMITED)
+        val data = MutableStateFlow(server.url("/red"))
+
+        composeTestRule.setContent {
+            val resId = data.collectAsState()
+            GlideImage(
+                data = resId.value.toString(),
+                modifier = Modifier.preferredSize(128.dp, 128.dp).testTag(GlideTestTags.Image),
+                onRequestCompleted = { loadCompleteSignal.offer(Unit) }
+            )
+        }
+
+        // Await the first load
+        runBlocking {
```

Comment:
What about making a `channel.await` ext fun mirroring latch to clarify testing code?

Reply:
Good idea. Added.